bundles/org.openhab.binding.froniuswattpilot/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,210 @@
+				<label>Hostname</label>
+				<description>Hostname or IP address of the device</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<context>password</context>
+				<label>Password</label>
+				<description>Password to access the device (the same password as required by the Solar.Wattpilot app)</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Channel Group Types -->
+	<channel-group-type id="control">
+		<label>Charging Control</label>
+		<channels>
+			<channel id="charging-allowed" typeId="charging-allowed"/>
+			<channel id="charging-mode" typeId="charging-mode"/>
+			<channel id="charging-current" typeId="charging-current"/>
+			<channel id="pv-surplus-threshold" typeId="pv-surplus-threshold"/>
+		</channels>
+	</channel-group-type>
+
+	<channel-group-type id="status">
+		<label>Charging Status</label>
+		<channels>
+			<channel id="charging-state" typeId="charging-state"/>
+			<channel id="charging-possible" typeId="charging-possible"/>
+			<channel id="single-phase" typeId="charging-single-phase"/>
+		</channels>
+	</channel-group-type>
+
+	<channel-group-type id="metrics">
+		<label>Charging Metrics</label>
+		<channels>
+			<channel id="power" typeId="system.electric-power">
+				<label>Total Power</label>
+			</channel>
+			<channel id="energy-session" typeId="system.electric-energy">
+				<label>Charged Energy</label>
+				<description>Energy charged in the current/last session</description>
+			</channel>
+			<channel id="energy-total" typeId="system.electric-energy">
+				<label>Total Charged Energy</label>
+				<description>Energy charged in total</description>
+			</channel>
+			<channel id="l1-power" typeId="system.electric-power">
+				<label>Phase 1 Power</label>
+			</channel>
+			<channel id="l2-power" typeId="system.electric-power">
+				<label>Phase 2 Power</label>
+			</channel>
+			<channel id="l3-power" typeId="system.electric-power">
+				<label>Phase 3 Power</label>
+			</channel>
+			<channel id="l1-voltage" typeId="system.electric-voltage">
+				<label>Phase 1 Voltage</label>
+			</channel>
+			<channel id="l2-voltage" typeId="system.electric-voltage">
+				<label>Phase 2 Voltage</label>
+			</channel>
+			<channel id="l3-voltage" typeId="system.electric-voltage">
+				<label>Phase 3 Voltage</label>
+			</channel>
+			<channel id="l1-current" typeId="system.electric-current">
+				<label>Phase 1 Current</label>
+			</channel>
+			<channel id="l2-current" typeId="system.electric-current">
+				<label>Phase 2 Current</label>
+			</channel>
+			<channel id="l3-current" typeId="system.electric-current">
+				<label>Phase 3 Current</label>
+			</channel>
+		</channels>
+	</channel-group-type>
+
+	<!-- Control Channel Types -->
+	<channel-type id="charging-allowed">
+		<item-type>Switch</item-type>
+		<label>Charging Allowed</label>
+		<description>Allow or forbid charging</description>
+		<category>BatteryLevel</category>
```

Comment:
I think `BatteryLevel` is intended for battery level channels, i.e. to visually show the charge level based on the channel state. Since this is a `Switch` channel, you might want to remove this category, or perhaps use `Battery` instead. Same for others below.